src/lib/pickles_types/plonk_verification_key_evals.mli

```diff
@@ -28,17 +86,73 @@ type 'comm t = 'comm Stable.Latest.t =
   ; mul_comm : 'comm
   ; emul_comm : 'comm
   ; endomul_scalar_comm : 'comm
+  ; optional_columns_comm : 'comm_opt Optional_columns.t
   }
 [@@deriving sexp, equal, compare, hash, yojson, hlist]
 
-val typ :
-     ('a, 'b, 'c) Snarky_backendless.Typ.t
-  -> ('a t, 'b t, 'c) Snarky_backendless.Typ.t
+(** Specialized instance of t for in-circuit representations. *)
+type ('a, 'bool) in_circuit = ('a, ('a, 'bool) Opt.t) t
+
+(** Specialized instance of t for out-of-circuit representations. *)
+type 'a out_circuit = ('a, 'a Option.t) t
 
-(** [map t ~f] applies [f] to all elements of type ['a] within record [t] and
-    returns the result. In particular, [f] is applied to the elements of
-    {!sigma_comm} and {!coefficients_comm}. *)
-val map : 'a t -> f:('a -> 'b) -> 'b t
+(** {2 Converters} *)
+
+(** [in_of_out oc_v] converts values from their out-of-circuit representation
+    [oc_v] to their in-circuit one. *)
+val in_of_out : 'a out_circuit -> ('a, 'b) in_circuit
```

Comment:
not sure if it would be any better, but have you considered `in_circuit_of_out_circuit`... a bit long maybe